db/puppies.js

```diff
@@ -0,0 +1,99 @@
+const client = require("./client");
+
+async function createPuppy({
+    name,
+    description,
+    breed,
+    weight,
+    size,
+    pedigree,
+    isVaccinated,
+    isNeutered,
+    gender,
+    isAvailable,
+    price
+}) {
+    try{
+        const { rows: [ puppy ] } = await client.query(`
+            INSERT INTO puppies(name, description, breed, weight, size, pedigree, isVaccinated, isNeutered, gender, isAvailable, price)
+            VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11)
+            RETURNING *;
+        `, [name, description, breed, weight, size, pedigree, isVaccinated, isNeutered, gender, isAvailable, price])
+    
+        return puppy;
+    } catch (error) {
+        console.error(error)
+    }
+}
+
+async function getAllPuppies() {
+    try{
+        const { rows } = await client.query(`
+            SELECT *
+            FROM puppies
+        `);
+    
+        return rows;
+    } catch (error) {
+        console.error(error)
+    }
+}
+
+async function getPuppyById(id) {
+    try{
+        const { rows: [ puppy ] } = await client.query(`
+            SELECT *
+            FROM puppies
+            WHERE id=$1;
+        `, [id])
+  
+        return puppy;
+    } catch (error) {
+        console.error(error)
+    }
+}
+
+async function updatePuppy({ id, ...fields }) {
+    try{
+        const setString = Object.keys(fields).map(
+            (key, index) => `"${ key }"=$${ index + 1 }`
+        ).join(', ');
+    
+        if (setString.length === 0) {
+            return;
+        }
+    
+        const { rows: [ puppy ] } = await client.query(`
+            UPDATE puppies
+            SET ${ setString }
+            WHERE id=${ id }
+            RETURNING *;
+        `, Object.values(fields));
+    
+        return puppy;
+    } catch (error) {
+        console.error(error)
+    }
+}
+
+async function deletePuppy(id) {
```

Comment:
I think deletePuppy should just set "isAvailable" field to false instead of deleting the puppy entirely since we still need the info stored for orders.

Reply:
I'm going to leave this function just in case there's a need to delete a puppy from the database, but leave it to the frontend to change isActive to false upon purchase